weblab/static/js/upload.js

```diff
@@ -58,6 +56,19 @@ Upload.prototype = {
     $('#dropbox a').click(function() {
       $("#fileupload").click();
     });
+
+    $("form #entityversionfilestable").on('click', 'tr .action .delete-file', function() {
+      var $tr = $(this).parents('tr');
+      $td = $tr.find(".filename");
+      if ($tr.hasClass('deleting')) {
+        $tr.removeClass('deleting');
+        $tr.find('input[name="delete_filename[]"]').remove()
+      } else {
+        $tr.addClass('deleting');
+        $('<input type="hidden" name="delete_filename[]" value="' + $.trim($td.text()) + '" />').appendTo($td);
+      }
+      return false;
+    });
   },
```

Comment:
Ah, that's OK because you don't show them the delete link for that file, just a 'cancel add' button effectively.

Interestingly, you can mark an existing file for deletion then upload a new file with the same name all within a single version change. Should we have a test that this does the right thing?

This is also possibly an unintended consequence of the fact that the check for a file upload having an existing name ignores files kept from previous versions - we should definitely fix that!

Reply:
I found a few bugs around marking for deletion and detecting existing uploaded files. It should now be possible to delete and replace a file in the same commit. (Might be useful if you have a local new copy of the file to upload - git will of course detect a deletion and re-addition as a change in the file)